raisinbread/test/api/LorisApiAuthenticatedTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+require_once __DIR__ .
+    "/../../../test/integrationtests/LorisIntegrationTest.class.inc";
+use GuzzleHttp\Client;
+
+/**
+ * PHPUnit class for API test suite. This script sends HTTP requests to every
+ * endpoints of the api module and look at the response content, status code and
+ * headers where it applies. All endpoints are accessible at <host>/api/<version>/
+ * (e.g. the endpoint of the version 0.0.3 of the API "/projects" URI for the host
+ * "example.loris.ca" would be https://example.loris.ca/api/v0.0.3/projects)
+ *
+ * @category   API
+ * @package    Tests
+ * @subpackage Integration
+ * @author     Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class LorisApiAuthenticatedTest extends LorisIntegrationTest
+{
+
+    protected $client;
+    protected $headers;
+    protected $base_uri;
+    protected $originalJwtKey;
+    protected $configIdJwt;
+ 
+    /**
+     * Call to LorisApiAuthenticationTest::setUp()
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        // Set up database wrapper and config
+        $this->factory = NDB_Factory::singleton();
+        $this->factory->reset();
+        $this->factory->setTesting(false);
+
+        $this->config = $this->factory->Config(CONFIG_XML);
+
+        $database = $this->config->getSetting('database');
+
+        $this->DB  = Database::singleton(
+            $database['database'],
+            $database['username'],
+            $database['password'],
+            $database['host'],
+            1
+        );
```

Comment:
```suggestion
        parent::setUp();
```

Reply:
That doesn't work. The JWT key has to be changed before `parent::setUp();` (or inside `LorisIntegrationTest.class.inc` setUp(), it but that's what was already done before ). If the key is changed after `parent::setUp()`, all test fail with `Unacceptable JWT key`, so this block is necessary.

Reply:
That is because the login function is overriden in AuthenticatedTest. You should rename your login function for something like apiLogin.
The original login function: https://github.com/aces/Loris/blob/main/test/integrationtests/LorisIntegrationTest.class.inc#L139